extended-tracer/README.md

```diff
@@ -0,0 +1,168 @@
+        //noinspection ConstantConditions
+        return carrier == null ? null : carrier.get(key);
+      }
+    };
+
+Map<String, String> normalizedTransport =
+    requestHeaders.entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                entry -> entry.getKey().toLowerCase(Locale.ROOT), Map.Entry::getValue));
+Context newContext = openTelemetry
+    .getPropagators()
+    .getTextMapPropagator()
+    .extract(Context.current(), normalizedTransport, TEXT_MAP_GETTER);
+try (Scope ignore = newContext.makeCurrent()) {
+  Span span = spanBuilder.setSpanKind(SERVER).startSpan();
+  try (Scope scope = span.makeCurrent()) {
+    transactionId = processCheckout(cartId);
+  } catch (Throwable e) {
+    span.setStatus(StatusCode.ERROR);
+    span.recordException(e);
+    throw e; // or throw new RuntimeException(e) - depending on your error handling strategy
+  } finally {
+    span.end();
+  }
+}
+```
+
+After:
+
+```java
+Tracing tracing = new Tracing(openTelemetry, "service");
+Map<String, String> propagationHeaders = tracing.getTextMapPropagationContext();
+// add propagationHeaders to request headers and call checkout service
+```
+
+```java
+// in checkout service: get request headers into a Map<String, String> requestHeaders
+Map<String, String> requestHeaders = new HashMap<>();
+String cartId = "cartId";
+
+Tracing tracing = new Tracing(openTelemetry, "service");
+String transactionId = tracing.traceServerSpan(requestHeaders,
+    tracer.spanBuilder("checkout_cart"), () -> processCheckout(cartId));
+```
+
+Note:
+
+- You can use `traceConsumerSpan` if you want to trace a consumer (e.g. from a message queue)
+  instead of a server.
+- Exceptions are re-thrown without modification - see [Exception handling](#exception-handling)
+  for more details.
+
+### Setting baggage entries
+
+Before:
+
+```java
+BaggageBuilder builder = Baggage.current().toBuilder();
+builder.put("key", "value");
+Context context = builder.build().storeInContext(Context.current());
+try (Scope ignore = context.makeCurrent()) {
+  String value = Baggage.current().getEntryValue("key");
+}
+```
+
+After:
+
+```java
+Tracing tracing = new Tracing(openTelemetry, "service");
+String value = Tracing.callWithBaggage(
+    Collections.singletonMap("key", "value"),
+    () -> Baggage.current().getEntryValue("key"))
+```
+
+## Exception handling
+
+`Tracing` re-throws exceptions without modification. This means you can catch exceptions around
+`Tracing` calls and handle them as you would without `Tracing`.
+
+Note that the `Tracing` methods do not declare any checked exceptions
+(the idea is taken from [@SneakyThrows](https://projectlombok.org/features/SneakyThrows)).
+Declaring a checked exception would force callers to handle it, which would create a lot of
+boilerplate code. Instead, `Tracing` re-throws checked exceptions as unchecked exceptions.
+
+## Component owners
+
+- [Gregor Zeitlinger](https://github.com/zeitlinger), Grafana Labs
```

Comment:
it would be great to get one more component owner, any takers?

Reply:
is this still applicable if the plan is to move this to core repo incubator?

Reply:
nope